src/System.Linq/src/System/Linq/ToCollection.cs

```diff
@@ -14,9 +14,18 @@ public static TSource[] ToArray<TSource>(this IEnumerable<TSource> source)
             {
                 throw Error.ArgumentNull(nameof(source));
             }
+            
+            if (source is TSource[] array)
+            {
+                return array;
+            }
```

Comment:
Right. FWIW the change can easily break code relying on existing shallow-copy behavior.